app/controllers/calculator_constants_controller.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class CalculatorConstantsController < ApplicationController
+  def index
+    @calculator_constants = CalculatorConstant.all
+  end
+
+  def update
+    updated_fields = []
+    params.each do |key, value|
+      update_calculator_constant(key, value, updated_fields)
+    end
```

Comment:
`params` is going to contain things like `"controller"=>"calculator_constants", "action"=>"update"`. I'd recommend you make keys & values be scoped to change this API to accept a single input param var, i.e.
```
"calculator_constants" => [
  {"name": "foo", "float_value": 999.9},
  {"name": "bar", "float_value": 999.9}
]
```

Reply:
Made the change. Also learned something about how to make an array of hashes for a form submit. Neat trick.